### DISCUSSION TITLE
Missing 'per Volume Statistics' in 24.02.0-1

### DISCUSSION THREAD

**Discussion by derDaywalker** [NetApp/harvest#2736] [General]
Hello everyone. We recently upgraded to the newest Version 24.02.0-1. Today we discovered, that Volume Dashboard doesn't provide Volume specific information anymore. To be more specific. Regularly we filter some vserver and volumes and want to see the physical / logical space usage and snapshot usage as well. This was found in the "per Volume Statsistics" graphs. But now there is only "Volume Capacity" which does only show top Resources and not specific Volumes which were selected in the top filter.

Hopefully the following picture shows what I mean:
Before:
![per_volume_stats](https://github.com/NetApp/harvest/assets/7250267/744187cb-d05d-4270-8ae9-b3bd85386201)

Now:
![volume_capacity](https://github.com/NetApp/harvest/assets/7250267/34ed7f3c-9469-4a27-9da8-b38a51948e83)

In the lower picture you can see, that the Volume Capacity Graphs show the top 5 Volumes and doesn't care about the selection. The titel "top 5" suggests, that this might be right, but if this is correct, where can the statistics for a selected volume be found?

At the same time the "Volume Performance" Graphs are also not consistent. 
![volume_performance](https://github.com/NetApp/harvest/assets/7250267/a3ab131f-71f8-4ab5-9c9a-3b539ccfca0d)

The title of each graph says "Top 5" but in fact only the selected volume is considered. Thus I guess there is a inconsistency. Either all the graphs should always only show the top volumes no matter if specific volumes are selected or the selected volumes should be used as a filter and not show the top volumes. 

The query for the volume_space_physical_used shows that the volumes are not considered (left side "volume capacity" query, right side "old per volume statistics"):
![image](https://github.com/NetApp/harvest/assets/7250267/d68141fe-93d1-49cf-af33-d91a30099c85)

On the other hand the "Volume Performance" from 24.02 uses the variable $Volume in its query:
![image](https://github.com/NetApp/harvest/assets/7250267/2f1cbb0f-79b2-4208-aab8-20c1eb6fccfd)

Thus for me the Volume Capacity queries are wrong (they should include the $Volume variable) and the Title of the graphs are not accurate because it says top 5 Volumes but in fact its like the "old per Volume Statistics". 

For a test we did rollback to Version 23.11.0-1. There everything was working as expected.

**Comment by rahulguptajss** [comment]
Thank you, @derDaywalker. Yes, this is a bug where the filter is missing. I agree that top k can be inconsistent when filters are selected and there are fewer than 'k' records in the results. I have opened an issue to fix the filtering #2741

**Reply by cgrinds** [reply]
hi @derDaywalker this is fixed. If you want to try these changes out before the next release, you can grab the [nightly build](https://github.com/NetApp/harvest/releases/tag/nightly). Alternatively, you can update your current dashboard with the fixed dashboard [json](https://github.com/NetApp/harvest/blob/main/grafana/dashboards/cmode/volume.json).

**Reply by derDaywalker** [reply]
Hi, thank you very much for the fast fix. We have updated our dashboard via the json file. This fixes our problem. Thus thanks again!

**Reply by cgrinds** [reply]
Appreciate the confirmation
